core/src/impl/Kokkos_HostSpace.cpp

```diff
@@ -308,6 +308,7 @@ void HostSpace::impl_deallocate(
     const size_t arg_alloc_size, const size_t arg_logical_size,
     const Kokkos::Tools::SpaceHandle arg_handle) const {
   if (arg_alloc_ptr) {
+    Kokkos::fence("HostSpace::impl_deallocate before free");
```

Comment:
In the long run, we'll want to be able to associate this with a particular execution space; e.g. if we have partitioned sets of CPU threads, we don't want to block on all of them completing in-flight work, rather than just the one(s) that were using the memory being deallocated.

Right now, this is fine.

Reply:
We discussed the scope of this fence elsewhere (I think it was on slack) and @crtrott was saying that he prefers a global fence over figuring out the execution spaces that can possibly access it.